### PR TITLE
[BUILD SYSTEM] Remove zstd

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Context:** We do not want to ship zstd.

**Description of the Change:** Removes zstd from LLVM build process.

**Benefits:** No zstd.

**Possible Drawbacks:** Possibly our bytecode takes more space.

**Related GitHub Issues:**

We will need to remove all caches for this change to take into effect.

### CODE REVIEW

**Comment by erick-xanadu** [comment]
https://github.com/PennyLaneAI/catalyst/actions/runs/11729023698 these wheels do not contain zstd. The cache must be purged for this change to take effect.

**Comment by rmoyard** [comment]
What was the reason to set it to `FORCE_ON`?

File: .github/workflows/build-wheel-linux-x86_64.yaml [hunk 1]
```diff
@@ -180,7 +180,7 @@ jobs:
               -DLLVM_ENABLE_ASSERTIONS=ON \
               -DLLVM_INSTALL_UTILS=ON \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-linux-x86_64.yaml:182]:
```suggestion
              -DLLVM_ENABLE_ZLIB=ON \
```

File: .github/workflows/build-wheel-linux-x86_64.yaml [hunk 2]
```diff
@@ -376,7 +376,7 @@ jobs:
               -DEnzyme_DIR=$GITHUB_WORKSPACE/enzyme-build \
               -DENZYME_SRC_DIR=$GITHUB_WORKSPACE/mlir/Enzyme \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-linux-x86_64.yaml:378]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/build-wheel-macos-arm64.yaml [hunk 3]
```diff
@@ -145,7 +145,7 @@ jobs:
               -DLLVM_ENABLE_ASSERTIONS=ON \
               -DLLVM_INSTALL_UTILS=ON \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-arm64.yaml:147]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/build-wheel-macos-arm64.yaml [hunk 4]
```diff
@@ -181,7 +181,7 @@ jobs:
               -DPython3_EXECUTABLE=$(which python${{ matrix.python_version }}) \
               -DLLVM_ENABLE_LLD=OFF \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-arm64.yaml:183]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/build-wheel-macos-arm64.yaml [hunk 5]
```diff
@@ -340,7 +340,7 @@ jobs:
               -DEnzyme_DIR=$GITHUB_WORKSPACE/enzyme-build \
               -DENZYME_SRC_DIR=$GITHUB_WORKSPACE/mlir/Enzyme \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-arm64.yaml:342]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/build-wheel-macos-x86_64.yaml [hunk 6]
```diff
@@ -141,7 +141,7 @@ jobs:
               -DLLVM_ENABLE_ASSERTIONS=ON \
               -DLLVM_INSTALL_UTILS=ON \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-x86_64.yaml:143]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/build-wheel-macos-x86_64.yaml [hunk 7]
```diff
@@ -177,7 +177,7 @@ jobs:
               -DPython3_EXECUTABLE=$(which python${{ matrix.python_version }}) \
               -DLLVM_ENABLE_LLD=OFF \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-x86_64.yaml:179]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/build-wheel-macos-x86_64.yaml [hunk 8]
```diff
@@ -331,7 +331,7 @@ jobs:
               -DEnzyme_DIR=$GITHUB_WORKSPACE/enzyme-build \
               -DENZYME_SRC_DIR=$GITHUB_WORKSPACE/mlir/Enzyme \
               -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/build-wheel-macos-x86_64.yaml:333]:
```suggestion
              -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/scripts/linux_arm64/rh8/build_catalyst.sh [hunk 9]
```diff
@@ -71,7 +71,7 @@ cmake -S mlir -B quantum-build -G Ninja \
     -DEnzyme_DIR=/catalyst/enzyme-build \
     -DENZYME_SRC_DIR=/catalyst/mlir/Enzyme \
     -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/scripts/linux_arm64/rh8/build_catalyst.sh:73]:
```suggestion
    -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: mlir/Makefile [hunk 10]
```diff
@@ -25,7 +25,7 @@ endif
 
 ENABLE_LLD?=$(DEFAULT_ENABLE_LLD)
 ENABLE_ZLIB?=OFF
```
Comment (erick-xanadu) [inline mlir/Makefile:27]:
```suggestion
ENABLE_ZLIB?=ON
```

File: .github/workflows/scripts/linux_arm64/rh8/build_llvm.sh [hunk 11]
```diff
@@ -42,7 +42,7 @@ cmake -S /catalyst/mlir/llvm-project/llvm -B /catalyst/llvm-build -G Ninja \
     -DLLVM_ENABLE_ASSERTIONS=ON \
     -DLLVM_INSTALL_UTILS=ON \
     -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/scripts/linux_arm64/rh8/build_llvm.sh:44]:
```suggestion
    -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

File: .github/workflows/scripts/linux_arm64/rh8/build_mhlo.sh [hunk 12]
```diff
@@ -48,7 +48,7 @@ cmake -S /catalyst/mlir/mlir-hlo -B /catalyst/mhlo-build -G Ninja \
     -DPython3_EXECUTABLE=/usr/bin/python3 \
     -DLLVM_ENABLE_LLD=ON \
     -DLLVM_ENABLE_ZLIB=OFF \
```
Comment (erick-xanadu) [inline .github/workflows/scripts/linux_arm64/rh8/build_mhlo.sh:50]:
```suggestion
    -DLLVM_ENABLE_ZLIB=FORCE_ON \
```

**Review by mlxd — APPROVED** [review]
Thanks Erick. No issue by me.
Do we know the size difference of the wheels with this change?

**Review by jay-selby — APPROVED** [review]
Looks good. Thanks Erick for handling this quickly

**Comment by erick-xanadu** [comment]
>  Do we know the size difference of the wheels with this change?

Looks like just 1 MB. It could have been more if we removed both zlib and zstd. But keeping zlib makes the difference very small.

**Pull request merged**